app/page.tsx

```diff
@@ -87,36 +99,78 @@ async function getMsgRolloutCollection(
 
 export default async function Dashboard() {
   // Check to see if Auth is enabled
-  const isAuthEnabled = process.env.IS_AUTH_ENABLED === 'true';
+  const isAuthEnabled = process.env.IS_AUTH_ENABLED === "true";
 
-  const recipeCollection = new NimbusRecipeCollection()
-  await recipeCollection.fetchRecipes()
-  console.log('recipeCollection.length = ', recipeCollection.recipes.length)
+  const recipeCollection = new NimbusRecipeCollection();
+  await recipeCollection.fetchRecipes();
+  console.log("recipeCollection.length = ", recipeCollection.recipes.length);
 
   // XXX await Promise.allSettled for all three loads concurrently
-  const localData = await getASRouterLocalMessageInfoFromFile()
-  const msgExpRecipeCollection = await getMsgExpRecipeCollection(recipeCollection)
-  const msgRolloutRecipeCollection = await getMsgRolloutCollection(recipeCollection)
+  const localData = await getASRouterLocalMessageInfoFromFile();
+  const msgExpRecipeCollection = await getMsgExpRecipeCollection(
+    recipeCollection
+  );
+  const msgRolloutRecipeCollection = await getMsgRolloutCollection(
+    recipeCollection
+  );
 
-  // get in format useable by MessageTable
-  const experimentAndBranchInfo : RecipeOrBranchInfo[] =
+  // Get in format useable by MessageTable
+  const experimentAndBranchInfo: RecipeOrBranchInfo[] = await Promise.all(
     msgExpRecipeCollection.recipes.map(
-      (recipe : NimbusRecipe) => recipe.getRecipeInfo())
+      async (recipe: NimbusRecipe): Promise<RecipeInfo> => {
+        let branches = await Promise.all(
+          recipe
+            .getBranchInfos()
+            .map(async (branch: BranchInfo): Promise<BranchInfo> => {
+              // We are making all branch ids upper case to make up for Looker being case sensitive
+              const ctrPercent = await setCTRPercent(branch.id.toUpperCase(), branch.template)
+              if (ctrPercent) {
+                branch.ctrPercent = ctrPercent
+              }
+
+              return branch;
+            })
+        );
+        // Update branches with CTR data for each recipe
+        let updatedRecipe = recipe.getRecipeInfo();
+        updatedRecipe.branches = branches;
+        return updatedRecipe;
+      }
+    )
+  );
 
-  const totalExperiments = msgExpRecipeCollection.recipes.length
+  const totalExperiments = msgExpRecipeCollection.recipes.length;
 
-  const msgRolloutInfo: RecipeOrBranchInfo[] =
+  const msgRolloutInfo: RecipeOrBranchInfo[] = await Promise.all(
```

Comment:
This looks (almost) identical to the rollout one, so it can be hoisted into its own function too, further de-duping the code and making it more testable.